rdkPlugins/DeviceMapper/source/DeviceMapper.cpp

```diff
@@ -0,0 +1,233 @@
+        {
+            deviceNodes.emplace_back(node);
+        }
+    }
+
+    // We know we have some devices we're interested in, but we didn't find any
+    // of them, so something's gone wrong
+    if (deviceNodes.size() == 0)
+    {
+        AI_LOG_FN_EXIT();
+        return false;
+    }
+
+    // Fix up the devices list
+    std::vector<DevNode> incorrectDevNodes;
+    for (size_t i = 0; i < mContainerConfig->linux->devices_len; i++)
+    {
+        auto configDevice = mContainerConfig->linux->devices[i];
+        std::string devicePath = configDevice->path;
+
+        // Is this device one we're interested in?
+        auto devIt = std::find_if(deviceNodes.begin(), deviceNodes.end(),
+                                [&p = devicePath](DevNode n)
+                                {
+                                    return p == n.path;
+                                });
+
+        if (devIt == deviceNodes.end())
+        {
+            continue;
+        }
+
+        // Check if the ids in config match the real ids
+        if (devIt->major == configDevice->major && devIt->minor == configDevice->minor)
+        {
+            // Config is correct, nothing to do
+            AI_LOG_DEBUG("No fixup needed for %s", devIt->path.c_str());
+            continue;
+        }
+
+        AI_LOG_INFO("Fixing major/minor ID for dev node '%s'", devIt->path.c_str());
+
+        // Track that we changed this node, as we'll refer to it later to
+        // fix up the cgroup allow list
+        DevNode tmp = {
+            devicePath,
+            devIt->major,
+            devIt->minor,
+            configDevice->major,
+            configDevice->minor,
+            devIt->mode
+        };
+        incorrectDevNodes.emplace_back(tmp);
+
+        // Edit the config
+        configDevice->major = devIt->major;
+        configDevice->minor = devIt->minor;
+    }
+
+    if (incorrectDevNodes.size() == 0)
+    {
+        // No more work to do
+        AI_LOG_FN_EXIT();
+        return true;
+    }
+
+    // Fix up the cgroup list
+    for (size_t i = 0; i < mContainerConfig->linux->resources->devices_len; i++)
+    {
+        auto configDev = mContainerConfig->linux->resources->devices[i];
+
+        // See if the device matches one we just fixed (have to compare ids as
+        // the dev path isn't in this section)
+        auto it = std::find_if(incorrectDevNodes.begin(), incorrectDevNodes.end(),
+                                [&cd = configDev](DevNode node)
+                                {
+                                    return (cd->major == node.configMajor && cd->minor == node.configMinor);
+                                });
+
+        if (it == incorrectDevNodes.end())
+        {
+            continue;
+        }
+
+        AI_LOG_INFO("Fixing major/minor ID in cgroup allow list for dev node '%s'", it->path.c_str());
+        configDev->major = it->major;
+        configDev->minor = it->minor;
+    }
```

Comment:
We are not checking if we handled every `incorrectDevNodes` entry. Maybe we should delete entry after fixing it and, after this whole loop, check for `incorrectDevNodes.size() != 0` again. It would mean someone had some issue with misspelling either device path or id, but it would have also the benefit of speeding computation in case of long lists of device to be mapped (which we both know no one will ever do *wink-wink*).